### PR TITLE
Anonymous IP TR Implementation

### PR DESCRIPTION
**Pull request by PeterRyder**
- Initial review of anonymous blocking in the TR
- Unit tests require the maxmind anonymous ip database which will not be committed to this repo - tests which require the database will be ignored if the file doesn't exist in the resources folder

### CODE REVIEW

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-traffic_ops-test-PR/14/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR-trafficops-test/17/

**Comment by dneuman64** [comment]
https://lists.apache.org/thread.html/35cd0620f84408e77bbdd5d650c83ee4edb5f87cd1bb08f7f397c7a1@%3Cdev.trafficcontrol.apache.org%3E

**Comment by knutsel** [comment]
Any idea when this feature will be merged? @PeterRyder ?

**Comment by PeterRyder** [comment]
I will update with a bug or 2 we found since I opened this. Will also make a separate PR for the Ops changes

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/722/
Test FAILed.

**Comment by blatimerse** [comment]
build failed on missing maven download -- triggering Jenkins to try again:

retest this please

**Comment by dangogh** [comment]
Jenkins build failed due to missing maven download -- triggering another build:

retest this please

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/723/
Test FAILed.

**Comment by dangogh** [comment]
and again...

retest this please

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/727/
Test FAILed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/728/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/729/
Test PASSed.

File: traffic_router/core/src/main/java/com/comcast/cdn/traffic_control/traffic_router/core/loc/AnonymousIpConfigUpdater.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2017 Cisco Systems, Inc.
```
Comment (limited) [inline traffic_router/core/src/main/java/com/comcast/cdn/traffic_control/traffic_router/core/loc/AnonymousIpConfigUpdater.java]: Please remove

File: traffic_router/core/src/main/java/com/comcast/cdn/traffic_control/traffic_router/core/loc/AnonymousIpDatabaseUpdater.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2017 Cisco Systems, Inc.
```
Comment (limited) [inline traffic_router/core/src/main/java/com/comcast/cdn/traffic_control/traffic_router/core/loc/AnonymousIpDatabaseUpdater.java]: Please remove

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/731/
Test FAILed.

**Comment by limited** [comment]
retest this please

**Comment by dneuman64** [comment]
@PeterRyder is this one ready to be tested too?

**Comment by PeterRyder** [comment]
Yes, I believe this is ready to be tested. @limited only had 1 comment which was to fix a license header

**Comment by dneuman64** [comment]
Sweet, do you think you could add some docs?

**Comment by PeterRyder** [comment]
Yes - I figured splitting this into 2 PRs would be easier to review

**Comment by PeterRyder** [comment]
Closing - refer to https://github.com/apache/incubator-trafficcontrol/pull/2184 for new PR with 2.3.x rebase changes

**Pull request closed without merging**